packages/iov-bns/src/bnscodec.spec.ts

```diff
@@ -14,8 +29,84 @@ import {
   swapClaimTxJson,
   swapOfferTxJson,
 } from "./testdata.spec";
+import { encodeBnsAddress } from "./util";
 
 describe("bnscodec", () => {
+  fit("generate test data", () => {
+    const nonces = [0 as Nonce, 1 as Nonce, Number.MAX_SAFE_INTEGER as Nonce] as const;
```

Comment:
I never tried property-based testing in JS/TS but might be worth looking into something like this: https://github.com/dubzzz/fast-check